libs/sdk-core/src/breez_services.rs

```diff
@@ -314,25 +323,39 @@ impl BreezServices {
         {
             Some(_) => Err(SendPaymentError::AlreadyPaid),
             None => {
-                self.persist_pending_payment(&parsed_invoice, amount_msat, req.label.clone())?;
-                let payment_res = self
-                    .node_api
-                    .send_payment(
-                        parsed_invoice.bolt11.clone(),
-                        req.amount_msat,
-                        req.label.clone(),
-                    )
-                    .map_err(Into::into)
-                    .await;
-                let payment = self
-                    .on_payment_completed(
-                        parsed_invoice.payee_pubkey.clone(),
-                        Some(parsed_invoice),
-                        req.label,
-                        payment_res,
-                    )
+                let pending_payment =
+                    self.persist_pending_payment(&parsed_invoice, amount_msat, req.label.clone())?;
+                self.notify_event_listeners(BreezEvent::payment_started(pending_payment.clone()))
                     .await?;
-                Ok(SendPaymentResponse { payment })
+
+                let pending_timeout_sec = req.pending_timeout_sec.unwrap_or(u64::MAX);
+                let (tx, rx) = std::sync::mpsc::channel();
+                let cloned = self.clone();
+                tokio::spawn(async move {
```

Comment:
Instead of using an async task and a channel perhaps we can simplify by using tokio::time::timeout ?
for example: 

`let result = tokio::time::timeout(Duration::from_seconds(pending_timeout), node_api.send_payment... )`

Reply:
I've set this PR to draft while we work on other things, @ok300 has a proposal to improve it

But regarding the tokio timeout, isn't the future cancelled if the timeout occurs? In our case the future has to complete with or without timeout

Reply:
I see, good point as we don't want to show error while still pending.

Reply:
Also we want the sync to complete in the background if pending is already returned and the Payment(Succeed/Failed) to be emitted

Reply:
I added #958 to fix some issues around that and improve handling.

It's using a combination of both your ideas:
- the channel (`tx`, `rx`)
- `tokio::time::timeout(Duration::from_secs(pending_timeout_sec), rx).await`